blog/2024-02-26-true-composable-templates.md

```diff
@@ -0,0 +1,104 @@
+authors:
+  - eyalk
+tags: [platform engineering, templates, terraform modules, winglang, wing cloud]
+image: https://github.com/winglang/docsite/assets/1727147/7c395c6f-9cf2-43c6-b4b1-6a5ba4d058bb
+hide_table_of_contents: true
+---
+
+![Composable Templates](https://github.com/winglang/docsite/assets/1727147/7c395c6f-9cf2-43c6-b4b1-6a5ba4d058bb)
+
+In the journey of software development, particularly for cloud-based projects, templates often 
+seem like the go-to solution for their promise of speed and simplicity. This notion was at the 
+forefront of a conversation I had with a Product Manager who doesn’t delve into coding. I had 
+shared with him about my workshop project, a blend of React, Vite, WebSockets, and a dynamic 
+counter, which had garnered positive attention for its effectiveness and ease of setup during 
+a workshop. I told him that the audience was able to create a simple but powerful setup that 
+worked locally and was deployable on AWS using terraform. I noted that if I would have tried to 
+achieve this without Wing, it would have taken me two weeks to get to the same point. However, 
+the Product Manager offered a perspective that prompted deep reflection. He acknowledged the 
+appeal of my project but pointed out that its functionalities could potentially be replicated 
+using templates in any technology. The main point of his argument was about differentiation: if 
+a template can encapsulate all that Wing offers, then what truly sets Wing apart? He didn’t stop 
+there. He added that templates could be designed with a layer of customization, allowing 
+developers to tweak them for specific needs. This layer would enable any developer to adapt the 
+template to their unique project requirements.
+
+#### Why are we talking about Templates?
+
+In today's platform engineering space, where companies are creating strong teams with a clear 
+mission to “empower developers for self service experience in ways that adhere to the enterprise 
+concern” everything looks like a nail and templates are hammer. Whether they come in a form of 
+terraform modules, or a set of off the shelf repos you can clone. I would even go far and observe 
+teams using solutions like Render.com where you pick up a templatized starter project and start 
+working with it. Templates are a very powerful tool with a very appealing time to value, no wonder 
+this product manager asked about Wing vs a template on any technology.
+
+#### Templates and Their Limits
+
+While templates offer a starting point, their predefined structure can sometimes box in creativity 
+and innovation. They’re great for getting off the ground quickly but may not always cater to the 
+nuanced needs of every project. The ability to customize a template does introduce flexibility, 
+but this can also introduce complexity, requiring developers to navigate and manipulate the 
+template's constraints to suit their specific objectives.
```

Comment:
The text feels a bit too abstract. Maybe lead with a concrete example

Reply:
One example I like to give is what happens if there's a security bug in the template and now we have to chase all copies of the template to fix this bug.